lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_workspace_runtime.rb

```diff
@@ -165,7 +168,7 @@ def instantiate(uri, user, root = nil)
         raise
       ensure
         @current.pop if pushed
-        pop_state_machine_info if obj && obj.state_machine? && self.root
+        pop_state_machine_info if is_state_machine && self.root
```

Comment:
Can this just be `obj.try(:state_machine?)` ?  Then you do not need to defined the `is_state_machine` variable.

If it is not defined on the object it will return `nil` and you still have the desired outcome.

Reply:
I know the code works as-is, but I'm not a fan of having the `is_state_machine` initially only set within 2 levels of if-blocks.  Would prefer setting `is_state_machine = false` after the `begin` statement and then only setting to `true` in the `if obj.state_machine?` check.